models/ac_generator.cpp

```diff
@@ -201,7 +201,7 @@ nest::ac_generator::pre_run_hook()
 void
 nest::ac_generator::update( Time const& origin, const long from, const long to )
 {
-  assert( to >= 0 and ( delay ) from < kernel().connection_manager.get_min_delay() );
+  assert( to >= 0 and ( long ) from < kernel().connection_manager.get_min_delay() );
```

Comment:
Since `get_min_delay()` now returns `long`, we don't need the cast here, which anyways does not make sense since `from` is already `long` (and if we needed it, it should be a proper `static_cast< long >(...)`. But here we can reduce to
```suggestion
  assert( to >= 0 and from < kernel().connection_manager.get_min_delay() );
```
The same applies for all similar lines, which are actually removed by #2616. I suggest to wait for that PR before doing anything more here.

Reply:
Actually, we shouldn't touch the casts in this PR, but in #2506 which is on hold until this PR is merged, right?

Reply:
Sure, but there (probably) are many more such casts which were not removed in #2616.